src/monattack.cpp

```diff
@@ -810,12 +835,12 @@ void mattack::spit_sap(monster *z, int index)
     if (dam <= 0) {
         return;
     }
-    if (g->u.uncanny_dodge() ) {
+    if( target->uncanny_dodge() ) {
         return;
     }
-    add_msg(m_bad, _("A glob of sap hits you!"));
-    g->u.deal_damage( z, bp_torso, damage_instance( DT_BASH, dam ) );
-    g->u.add_effect("sap", dam);
+    add_msg( msg_type, _("A glob of sap hits %s!"), target->disp_name().c_str() );
```

Comment:
`target->add_msg_player_or_npc` would allow to keep the "a sap hits you!" part and includes the can_see for the targe.